satellite/satellitedb/overlaycache.go

```diff
@@ -167,65 +225,26 @@ func (cache *overlaycache) GetNodesNetwork(ctx context.Context, nodeIDs []storj.
 	return nodeNets, Error.Wrap(rows.Err())
 }
 
-func (cache *overlaycache) queryNodes(ctx context.Context, excludedNodes []storj.NodeID, count int, safeQuery string, args ...interface{}) (_ []*overlay.SelectedNode, err error) {
-	defer mon.Task()(&ctx)(&err)
-
-	if count == 0 {
-		return nil, nil
-	}
-
-	safeExcludeNodes := ""
-	if len(excludedNodes) > 0 {
-		safeExcludeNodes = ` AND id NOT IN (?` + strings.Repeat(", ?", len(excludedNodes)-1) + `)`
-		for _, id := range excludedNodes {
-			args = append(args, id.Bytes())
-		}
-	}
+func buildSelection(ctx context.Context, count int, safeQuery string) (query string, args []interface{}) {
```

Comment:
The query is returned even when count is 0.

Reply:
That should be fine: it ends up with `LIMIT 0` which lets the planner know that this part of the query can be ignored, and avoids the need to have a logic in the calling function to decide how to combine the two parts of the query (`UNION ALL` or not). And the execution time to build the query here is most likely dwarfed by the time it takes to execute the query. But it could be done either way, if you feel strongly about it.

Reply:
The benchmark does not support that it's completely ignored. Maybe there's something else with the query, but there's a time increase for SelectStorageNodes and SelectNewStorageNodes in the parallel case.

Reply:
You're right. I went back to verify. The planner doesn't ignore a subquery with `LIMIT 0`. I'm surprised, but there it is. Looks like we should indeed elide these when count=0.